src/materials/ShaderMaterial.js

```diff
@@ -47,6 +47,7 @@ THREE.ShaderMaterial = function ( parameters ) {
 	this.fog = false; // set to use scene fog
 
 	this.lights = false; // set to use scene lights
+	this.userClippingPlanes = false; // set to use user-defined clipping planes
```

Comment:
It's built it in analogy to `ShaderMaterial.lights`: When you write a custom shader you state that you would like to get the uniform passed in - IOW it says whether your custom shader supports clipping.

I don't like this stuff so much either (am tired of ranting about it), been following what appears to be common practice, and would find it better to replace code like

```javascript
if ( material instanceof ThatMaterial ||
        material instanceof AnotherMaterial ||
        material instanceof YetAnotherMaterial ||
        material.supportsMyFeature ) {

      // feed uniform

}
```
with
```javascript
if ( material.supportsMyFeature ) {

    // feed uniform

}
```
and define `supportsMyFeature` on the prototypes of the materials - unless for the ones wrapping custom shaders where the property gets set on the instance.